lib/src/editor.dart

```diff
@@ -0,0 +1,788 @@
+
+  /// Style configuration settings for [YamlEditor] when modifying the YAML string.
+  final YamlStyle defaultStyle;
+
+  @override
+  String toString() => _yaml;
+
+  YamlEditor(this._yaml, {this.defaultStyle = const YamlStyle()}) {
+    _contents = loadYamlNode(_yaml);
+  }
+
+  /// Returns the [YamlNode] present at the path. The [YamlNode] that is returned represents
+  /// the current value when the function is called, and will not be updated when the YAML
+  /// is updated in the future. For example,
+  ///
+  /// ```dart
+  /// final node = doc.parseValueAt(['YAML']);
+  ///
+  /// print(node.value); /// Expected output: "YAML Ain't Markup Language"
+  ///
+  /// doc.setIn(['YAML'], 'YAML');
+  ///
+  /// final newNode = doc.parseValueAt(['YAML']);
+  ///
+  /// print(newNode.value); /// "YAML"
+  /// print(node.value); /// "YAML Ain't Markup Language"
+  /// ```
+  ///
+  /// An [ArgumentError] will be thrown if calling the `[]` operator would have resulted in an
+  /// error, but `null` (as opposed to [YamlScalar] null) will be returned if the operation would
+  /// have resulted in a `null` value on a dart collection.
+  ///
+  /// ```dart
+  /// final doc = YamlEditor('{a: {d: 4}, c: ~}');
+  /// doc.parseValueAt(['b', 'd']); // ArgumentError
+  /// doc.parseValueAt(['b']); // null
+  /// doc.parseValueAt(['c']); // YamlScalar(null)
+  ///
+  /// final doc2 = YamlEditor('[0,1]');
+  /// doc2.parseValueAt([2]); // ArgumentError
+  /// doc2.parseValueAt(["2"]); // ArgumentError
+  /// ```
+  YamlNode parseValueAt(Iterable<Object> path) {
+    dynamic current = _contents;
+
+    for (var key in path) {
+      try {
+        current = current.nodes[key];
+      } catch (Error) {
+        throw ArgumentError(
+            'Invalid path $path: Invalid key $key supplied to $current');
+      }
+    }
+
+    return current;
+  }
+
+  /// Sets [value] in the [path]. Takes an optional [style] parameter.
+  ///
+  /// If the [path] is not accessible (e.g. it currently does not exist in the document),
+  /// an error will be thrown. Note that [setIn] provides a different result as compared to
+  /// a [removeIn] followed by an [insertIn], because it preserves comments at the same level.
+  ///
+  /// ```dart
+  /// final doc = YamlEditor('''
+  ///   - 0
+  ///   - 1 # comment
+  ///   - 2
+  /// ''');
+  /// doc.setIn([1], 'test');
+  /// ```
+  ///
+  /// Expected Output:
+  /// '''
+  ///   - 0
+  ///   - test # comment
+  ///   - 2
+  /// '''
+  ///
+  /// ```dart
+  /// final doc2 = YamlEditor("[YAML Ain't Markup Language   # comment]");
+  /// doc2.removeIn([1]);
+  /// doc2.insertInList([1], 'test');
+  /// ```
+  ///
+  /// Expected Output:
+  /// '''
+  ///   - 0
+  ///   - test
+  ///   - 2
+  /// '''
+  void setIn(Iterable<Object> path, Object value, {YamlStyle style}) {
+    final collectionPath = path.take(path.length - 1);
+    final maybeCollection = parseValueAt(collectionPath);
+    final lastNode = path.last;
```

Comment:
nit: it would probably be better to call this variable `keyOrIndex` or just `key`... or `targetKey` or `targetKeyOrIndex`
or `indexOrKey` whatever is prettiest..